### PR TITLE
 Modifies the rendering of constant values in rustdoc.

### PR DESCRIPTION
**Pull request by sladyn98**
Previously, non-literal expressions (like `50 + 50`) would be printed
  alongside their value as a comment (like ` = _; // 100i32`). This approach
  often resulted in confusing documentation, especially for complex expressions.

 To improve clarity, this commit changes the rendering to display the evaluated
  value of the expression directly (like ` = 100i32;`) unless the constant is a
  literal expression. The original expression is still included as a comment if it
  doesn't match the evaluated value.

  These changes aim to improve the readability and usefulness of constant value
  documentation generated by rustdoc.

### CODE REVIEW

**Comment by sladyn98** [comment]
CC @GuillaumeGomez I thought of implementing this fix before migration to Askama

**Comment by GuillaumeGomez** [comment]
This sounds like a good improvement to me. However, this is a very complicated topic. So first let's hear from the last person who worked (a lot) on this. cc @fmease

File: src/librustdoc/html/render/print_item.rs [hunk 1]
```diff
@@ -1474,22 +1474,17 @@ fn item_constant(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, c: &cle
...
-        //            `50 + 50`
-        //        which looks just wrong.
-        //        Should we print
-        //            ` = 100i32;`
-        //        instead?
-
         let value = c.value(tcx);
         let is_literal = c.is_literal(tcx);
         let expr = c.expr(tcx);
         if value.is_some() || is_literal {
```
Comment (clubby789) [inline src/librustdoc/html/render/print_item.rs:1480]: I think this surrounding it should be removed - the `else` will never be hit right now

**Review by fmease — CHANGES_REQUESTED** [review]
Overall the changes look fine, thanks for tackling the FIXME!
However please address the nits and update the tests.

For context, I wrote this FIXME and it would also be fixed by #99688
which is currently blocked on an RFC that I need to write (but which
I didn't do yet since I'm no longer sure if it's the right direction for
rustdoc but that's another story).

Your changes seem to be a good middle ground in the meantime.

File: src/librustdoc/html/render/print_item.rs [hunk 2]
```diff
@@ -1474,22 +1474,17 @@ fn item_constant(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, c: &cle
...
-            write!(w, " = {expr};", expr = Escape(&expr));
-        } else {
-            w.write_str(";");
+            if is_literal {
+                write!(w, " = {expr};", expr = Escape(&expr));
+            } else if let Some(value) = &value {
```
[the remark this answers is not in the document]
Comment (sladyn98) [reply src/librustdoc/html/render/print_item.rs]: Yeah that makes sense, removed the branch on lines 1490-1491

File: src/librustdoc/html/render/print_item.rs (hunk 1 above)
Comment (sladyn98) [reply src/librustdoc/html/render/print_item.rs]: Yeah perfect. Got rid of the outer if

**Pull request closed without merging**